tfhe-zk-pok/src/proofs/pke.rs

```diff
@@ -301,28 +329,29 @@ where
             c_hat,
             c_y,
             pi,
-            c_hat_t,
-            c_h,
-            pi_kzg,
+            optional_fields,
         } = compressed;
 
         Ok(Proof {
             c_hat: G::G2::uncompress(c_hat)?,
             c_y: G::G1::uncompress(c_y)?,
             pi: G::G1::uncompress(pi)?,
-            c_hat_t: c_hat_t.map(G::G2::uncompress).transpose()?,
-            c_h: c_h.map(G::G1::uncompress).transpose()?,
-            pi_kzg: pi_kzg.map(G::G1::uncompress).transpose()?,
-        })
-    }
-}
 
-impl<G: Curve> Proof<G> {
-    pub fn content_is_usable(&self) -> bool {
```

Comment:
we probably need a check that the points are properly on the elliptic curve, cc @nsarlin-zama, this was a function checking "integrity" of the proof in the sense that the raw data is valid EC data

Reply:
I don't think this was done here but I agree this should be part of the conformance for the zk. It can be done by calling `if self.is_on_curve() && self.is_in_correct_subgroup_assuming_on_curve()` on the Affine representation

Since I think this was quite heavy maybe it should be an option in the conformance params

Reply:
But it might be worthy of a dedicated PR

Reply:
agreed, I'm not sure I agree it should be an option for conformance however, as a wrong point during verification is likely a very bad idea

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/809

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/744